dysql/mappers.py

```diff
@@ -128,21 +128,47 @@ class RecordCombiningMapper(BaseMapper):
     unique ``id`` value.
     """
 
-    def __init__(self, record_mapper: Optional[Type[DbMapResultBase]] = DbMapResult):
+    def __init__(self, record_mapper: Optional[Type[DbMapResultBase]] = DbMapResult, id_columns: List[str] = None):
+        """
+        :param id_columns: id columns can be one or more columns that should uniquely identify a record so when we are
+        adding to lists when we expect a list of data for a single record.
+                    * Defaults to the 'id' column if nothing else is specified.
+                    * If each record should be a new record, pass in an empty array
+        :param record_mapper: what we are mapping records to.
+        """
+        self.id_columns = ['id']
+        if id_columns is not None:
+            self.id_columns = id_columns
         self.record_mapper = record_mapper
 
+    def _get_lookup(self, record):
+        """
+        Return a lookup key based on a set of defined columns
+        :param record:
+        :return:
+        """
+        if not self.id_columns:
+            return None
+        values = []
+        for column in self.id_columns:
+            if column not in record:
+                # returning none because we don't want to make assumptions about how to handle a complex key
+                return None
```

Comment:
Should this be a return here or just a `continue` so other keys can be considered?

Reply:
I don't want to make the assumption of a valid record if something is wrong with the 'unique' key the use is trying to define. you'd suggested it earlier but maybe raising an exception would be better here...

Reply:
Oh, I thought you were still going to use other columns though. i.e. `['name', 'setp']` for the id columns could still group on name, but not on `setp` since that column doesn't exist. I think what you have is fine, as long as we issue a warning in the log for it.

Reply:
I like the idea of a warning, I'll add it.